File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        n = len(blocks)
        whites = 0

        for i in range(0, k):
            if blocks[i] == 'W':
                whites += 1
        
        count = whites
        
        for i in range(k, n):
            if blocks[i - k] == 'W':
                whites -= 1
            if blocks[i] == 'W':
                whites += 1
            count = whites if whites < count else count
        
        return count

File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int n = blocks.length();
        int whites = 0;

        for (int i = 0; i < k; i++) {
            if (blocks.charAt(i) == 'W') {
                whites++;
            }
        }

        int res = whites;

        for (int i = k; i < n; i++) {
        if (blocks.charAt(i - k) == 'W') {
                whites--;
            }
            if (blocks.charAt(i) == 'W') {
                whites++;
            }

            res = Math.min(whites, res);
        }

        return res;
    }
}